funnel/static/js/scripts.js

```diff
@@ -1,80 +1,299 @@
-        return false;
-    });
+    Ractive.DEBUG = false;
 
-    $(".comments .comment-reply").click(function() {
-        var cfooter = $(this).parent();
-        $('#newcomment input[name="parent_id"]').val(cfooter.attr('data-id'));
-        $('#newcomment input[name="comment_edit_id"]').val('');
-        $("#toplevel-comment").removeClass('hidden');
-        $("#comment-submit").val("Reply"); // i18n gotcha
-        cfooter.after($("#newcomment"));
-        $("#newcomment textarea").focus();
-        return false;
+    this.participantTableContent = new Ractive({
+      el: '#participants-table-content',
+      template: '#participant-row',
+      data: {
+        participants: '',
+        getCsrfToken: function() {
+          return $('meta[name="csrf-token"]').attr('content');
+        },
+        getBadgeUrl: function(pid) {
+          return config.badgeUrl.replace('participant-id',pid);
+        },
+        getEditUrl: function(pid) {
+          return config.editUrl.replace('participant-id',pid);
+        },
+        getCheckinUrl: function() {
+          return config.checkinUrl;
+        }
+      }
+    });
+  },
+  update: function() {
+    var participantTable = this;
+    
+    $.ajax({
+      type: 'GET',
+      url: participantTable.Config.participantlistUrl,
+      timeout: 5000,
+      dataType: 'json',
+      success: function(data) {
+        participantTable.participantTableContent.set('participants', data.participants).then(function() {
+          $('.js-loader').hide();
+          $('.js-total').text(data.total_participants);
+          $('.js-totalcheckin').text(data.total_checkedin);
+          if (Modernizr.localstorage) {
+            var participants = tohashMap(data.participants, "pid");
+            participantTable.checkinQ.updateQueue(participants);
+            participantTable.cancelcheckinQ.updateQueue(participants);
+          }
+        });
+        $('.footable').trigger('footable_redraw');
+      }
     });
+  },
+  handleCheckIn: function() {
+    var participantTable = this;
 
-    $("#toplevel-comment a").click(function() {
-        $('#newcomment input[name="parent_id"]').val('');
-        $('#newcomment input[name="comment_edit_id"]').val('');
-        $("#comment-submit").val("Post comment"); // i18n gotcha
-        $(this).parent().after($("#newcomment"));
-        $(this).parent().addClass('hidden');
-        $("#newcomment textarea").focus();
-        return false;
+    $('#participants-table-content').on('submit', '.checkin_form', function(event) {
+      event.preventDefault();
+      var formValues = {};
+      $(this).serializeArray().map(function(obj) {
+        formValues[obj.name] = obj.value;
+      });
+      if (formValues["checkin"] === "t") {
+        participantTable.checkinQ.enqueue(formValues["pid"]);
+      }
+      else {
+        participantTable.cancelcheckinQ.enqueue(formValues["pid"]);
+      }
+      $(this).find('.js-loader').show();
```

Comment:
Wouldn't it be simpler to have the `js-loader` show or hide based on some state variable eg: `loading`? That would relieve us from explicitly showing and hiding the loader.